mlflow/keras.py

```diff
@@ -400,6 +394,13 @@ def on_train_end(self, logs=None):
                     type(self.model.optimizer.epsilon) is float \
                     else keras.backend.eval(self.model.optimizer.epsilon)
                 try_mlflow_log(mlflow.log_param, 'epsilon', epsilon)
+
+        def on_epoch_end(self, epoch, logs=None):
+            if not logs:
+                return
+            try_mlflow_log(mlflow.log_metrics, logs, step=epoch)
+
+        def on_train_end(self, logs=None):
```

Comment:
You should be safe to move the summary related logging to on_train_begin with the other parameters

Reply:
Done! Thanks